src/OrchardCore/OrchardCore.Modules/ModularTenantContainerMiddleware.cs

```diff
@@ -64,6 +67,8 @@ public class ModularTenantContainerMiddleware
                                 {
                                     tenantEvent.ActivatedAsync().Wait();
```

Comment:
It would be nice to remove the blocking call. Could the `lock` be replaced by a `SemaphoreSlim`? (or `AsyncLock` if you use Nito.AsyncEx)

Reply:
First, about the lock, it was already there. I don't retrieve the comment saying that the shell should be activated by only one request and synchronously, i mean another request needs to wait.

@sebastienros if this lock necessary? I think it is.
Then, can i try to use the Nito async lirary?

Reply:
@sebastienros @PinpointTownes i did something with `SemaphoreSlim`.

But i needed to have one per shellContext to not lock incoming request for another tenant. So, i did something where the shellContext embeds a semaphore. But need to be reviewed because you know that i can do terrific code ;) So, see my last commit, it's only a suggestion.